### DISCUSSION TITLE
Source Organization

### DISCUSSION THREAD

**Discussion by jamierytlewski** [mkosir/typescript-style-guide#5] [General]
Why do each of the components have index.tsx for the main component, but then the `component.stories.tsx` and `component.test.tsx` start with the component/folder name. I would think it would be better to have 

```
  ProductItem/
  ├─ ProductItem.tsx
  ├─ ProductItem.stories.tsx
  └─ ProductItem.test.tsx
```

**Comment by mkosir** [comment]
Hey @jamierytlewski, I agree with you, just don't like the importing path as `./ProductItem/ProductItem`.
We could add another `index.ts` file as:
  ```
  ProductItem/
  ├─ index.tsx
  ├─ ProductItem.tsx
  ├─ ProductItem.stories.tsx
  └─ ProductItem.test.tsx
  ```
  which would serve as barrel export and allow us to export/expose only what we want (component, types etc.), wdyt?

**Reply by jamierytlewski** [reply]
This makes a lot of sense to me.

**Reply by mkosir** [reply]
Any contribution is much appreciated 🙇‍♂️, feel free to come up with PR!
